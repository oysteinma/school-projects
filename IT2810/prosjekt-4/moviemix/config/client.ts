import { ApolloClient, NormalizedCacheObject } from "@apollo/client";

import { cache } from "./cache";

/**
 * Setup Apollo Client
 */
export const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
  cache: cache,
  uri: "https://it2810-59s.idi.ntnu.no/graphql",
});
