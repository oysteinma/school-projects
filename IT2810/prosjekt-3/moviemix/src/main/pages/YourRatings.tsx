import { useQuery } from "@apollo/client";
import Loading from "../components/Loading";
import ErrorMessage from "../components/ErrorMessage";
import { GET_RATINGS_BY_USER } from "../queries/query";
import MovieMixHeader from "../components/MovieMixHeader";
import { Box } from "@mui/material";
import ListView from "../components/ListView";

/**
 *
 * @returns the YourRatings page, which displays the movies the user has rated
 * or none if the user has not rated any movies
 */
export default function YourRatings() {
  const { loading, error, data } = useQuery(GET_RATINGS_BY_USER, {
    variables: {
      id: window.localStorage.getItem("userId"),
    },
    fetchPolicy: "network-only",
  });

  if (loading) return <Loading />;
  if (error) return <ErrorMessage {...error} />;

  return (
    <Box
      display={"flex"}
      flexDirection={"column"}
      alignItems={"center"}
      alignContent={"center"}
      width={"50vw"}
      mx={"auto"}
      marginTop={"5vh"}
      marginBottom={"5vh"}
      gap={"5vh"}
    >
      <MovieMixHeader title="Your Ratings" />
      <ListView {...data} />
    </Box>
  );
}
