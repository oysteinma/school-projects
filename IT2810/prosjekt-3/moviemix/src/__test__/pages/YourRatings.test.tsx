import YourRatings from "../../main/pages/YourRatings";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { HashRouter as Router } from "react-router-dom";

/**
 * @tests The your ratings page component - namely that the component renders
 */

it("check if YourRatings renders succesfully", () => {
  const { container } = render(
    <MockedProvider>
      <Router>
        <YourRatings />
      </Router>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});
