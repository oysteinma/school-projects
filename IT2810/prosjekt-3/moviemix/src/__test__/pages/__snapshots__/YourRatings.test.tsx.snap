// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if YourRatings renders succesfully 1`] = `
<div>
  <div
    class="MuiBox-root css-z7qmyv"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 5rem; height: 5rem;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
